core/src/replay_stage.rs

```diff
@@ -3089,15 +3090,40 @@ impl ReplayStage {
                     (bank.slot(), bank.hash()),
                     Some((bank.parent_slot(), bank.parent_hash())),
                 );
+
                 bank_progress.fork_stats.bank_hash = Some(bank.hash());
-                let bank_frozen_state = BankFrozenState::new_from_state(
+                let mut bank_frozen_state = BankFrozenState::new_from_state(
                     bank.slot(),
                     bank.hash(),
                     duplicate_slots_tracker,
                     duplicate_confirmed_slots,
                     heaviest_subtree_fork_choice,
                     epoch_slots_frozen_slots,
                 );
+
+                if bank
+                    .feature_set
+                    .is_active(&solana_sdk::feature_set::vote_only_full_fec_sets::id())
+                    // No reason to check our leader block
+                    && bank.collector_id() != my_pubkey
```

Comment:
`set-identity` sanity check:
this should be fine as we do not update `my_pubkey` between `maybe_start_leader` and `replay_active_banks`. In fact we already have a `my_pubkey` comparison happening in `replay_active_bank` https://github.com/anza-xyz/agave/blob/12d009ececd7190a9f9f940715832001c03be2f4/core/src/replay_stage.rs#L2926